content/reference/wadm/api.md

```diff
@@ -0,0 +1,252 @@
+
+Deploys the indicated model. This operation is idempotent, and will simply "succeed" if a request to deploy an already deployed model is received. The previous version, if deployed, does not need to be _undeployed_. Instead, the desired state of the new version is compared against the observed state of the lattice and the autonomous agents will compensate accordingly.
+
+**Request**:
+```json
+{
+    "version": "..version to deploy"
+}
+```
+
+**Response**:
+```json
+{
+    "result": "success|error",
+    "message": "..."
+}
+```
+
+It's important to note that the success/fail of this call doesn't indicate completed deployment, only success or failure of the receipt of the deployment request. In other words, the response is an _ack_.
+
+### Undeploy
+`wadm.api.{lattice}.model.undeploy.{name}`
+
+Undeploys the currently deployed version of a model. If the model is already undeployed, the operation is idempotent and will return success along with an appropriate message. If the request field `destructive` is true, then the agent will attempt to remove all resources managed by the deployment from the lattice. Otherwise, previously managed resources will remain running. 
+
+**Note** The response is delivered immediately as an acknowledgement, as it can take several minutes to complete a "destructive" undeploy.
+
+**Request**: 
+```json
+{
+    "destructive": false
+}
+```
+
+**Response**:
+```json
+{
+    "result": "success|error",
+    "message": "..."
+}
+```
+
+### Deployment Status
+`wadm.api.{lattice}.model.status.{name}`
+
+Describes the deployment status of a given model. Note that since only one version of a model is ever deployed at one time, we don't need to specify a version here. The currently deployed version is included in the response.
+
+The response includes a `status` field which is an aggregate at high levels and an individual status for single traits.
+
+**Request**: `none`
+
+**Response**:
+```json
+{
+    "version": "...",
+    "status": 
+        {
+            "type": "undeployed|compensating|ready|failed",
+            "message": "..."
+        },
+    "components": [
+        {
+            "name": "...",
+            "type": "...",
+            "status": {
+                "type": "...",
+                "message": "..."
+            },
+            "traits": [
+                {
+                    "name": "...",
+                    "type": "spreadscaler|linkdef",
+                    "status": {
+                        "type": "...",
+                        "message": "..."
+                    }
+                }
+            ]
+        }
+    ]
+}
+```
+
+### Deployment History
+`wadm.api.{lattice}.model.history.{name}`
+
+Returns the history of compensating actions taken by wadm. Also includes when deploy and undeploy actions were taken.
+
+**Request**: `none`
+
+**Response**:
+```json
+[
+    {
+        "time" : "<RFC3339 Timestamp>",
+        "action": "modeldeploy|modelundeploy|compensator",
```

Comment:
could change 'modeldeploy' and 'modelundeploy' to 'deploy' and 'undeploy' to match the operation names